connectivity/drivers/wifi/esp8266-driver/mbed_lib.json

```diff
@@ -92,7 +92,10 @@
         }
     },
     "target_overrides": {
-         "HEXIWEAR": {
+        "*": {
+            "target.printf_lib": "std"
```

Comment:
Compilation error:
```
[ERROR] Attempt to override undefined parameter 'target.printf_lib' in 'library:esp8266[*]'
````
https://github.com/ARMmbed/mbed-os/pull/13707/files#diff-72daba8ef643af857ee7d9ec1b681b23R96

Reply:
I'm sorry, this is embarassing, it would've helped if I compiled for a target that actually uses esp8266

Reply:
You can use the reproducing steps in https://github.com/ARMmbed/mbed-os/issues/13687 issue description: